src/serviceFacades/apps.js

```diff
@@ -314,6 +316,23 @@ function adminUpdateApp({ app, details, avus, values }) {
         );
     }
 
+    if (app.isBlessed !== values.isBlessed) {
+        promises.push(
+            adminAddAVUToApp({
+                appId: app.id,
+                avu: {
+                    avus: [
+                        {
+                            attr: BLESSED_ATTR,
+                            value: values.isBlessed.toString(),
+                            unit: "",
+                        },
+                    ],
+                },
+            })
```

Comment:
Is this really what the service is expecting, in terms of how the listing will set the `isBlessed` flag?

It works differently for beta AVUs, where the `isBeta` flag is `true` if the app has the beta AVU at all, and `false` if the beta AVU is not included in the metadata's `avus` list.

Also, the request looks odd to me. I would expect it to look like either `{appId: app.id, avu: { attr: BLESSED_ATTR, value, unit }}` or the way it is now, but with `metadata` instead of `avu`: `{appId: app.id, metadata: { avus: [{ attr: BLESSED_ATTR, value, unit }] }}`

Reply:
Hmm am not familiar with API side of things. I `POST` to `http://localhost:3000/api/admin/apps/67d15627-22c5-42bd-8daf-9af5deecceab/metadata` (for DE word count app in QA) and I get 
`{
   "avus":[
      {
         "modified_on":1617135897846,
         "unit":"",
         "value":"true",
         "target_id":"67d15627-22c5-42bd-8daf-9af5deecceab",
         "created_on":1617135897846,
         "id":"fef5ae84-9195-11eb-874a-62d47aced14b",
         "avus":[
            
         ],
         "modified_by":"sriram",
         "created_by":"sriram",
         "attr":"cyverse-blessed"
      },
      {
         "modified_on":1468953122554,
         "unit":"",
         "value":"http://edamontology.org/operation_2409",
         "target_id":"67d15627-22c5-42bd-8daf-9af5deecceab",
         "created_on":1468953122554,
         "id":"161ef66a-4ddf-11e6-a3bb-008cfa5ae621",
         "avus":[
            {
               "modified_on":1468953122554,
               "unit":"value",
               "value":"Data handling",
               "target_id":"161ef66a-4ddf-11e6-a3bb-008cfa5ae621",
               "created_on":1468953122554,
               "id":"161f2c7a-4ddf-11e6-a3bb-008cfa5ae621",
               "avus":[
                  
               ],
               "modified_by":"rogerab",
               "created_by":"rogerab",
               "attr":"rdfs:label"
            }
         ],
         "modified_by":"rogerab",
         "created_by":"rogerab",
         "attr":"rdf:type"
      },
      {
         "modified_on":1566410803437,
         "unit":"",
         "value":"iplant:de:prod:communities:Mescal",
         "target_id":"67d15627-22c5-42bd-8daf-9af5deecceab",
         "created_on":1566410803437,
         "id":"6e9e60ae-c43e-11e9-a03c-008cfa5ae621",
         "avus":[
            
         ],
         "modified_by":"sarahr",
         "created_by":"sarahr",
         "attr":"cyverse-community"
      }
   ]
}`

Reply:
App seems to be marked as blessed right away.

Reply:
Ah, I double checked `adminAddAVUToApp`, and it looks like it's expecting the `avu` key, but passing that as the body of the metadata request (guess I missed this in #263).

What about posting `{"avus":[{"attr":"cyverse-blessed","value":"false","unit":""}]}`?
Will that still set the `isBlessed` flag in the app listing?

I also suggest renaming the `avu` key to `metadata` in `adminAddAVUToApp`. Otherwise it's confusing because each AVU can have it's own sub-listing of `avus` (and this request looks like it's trying to add 1 avu with a sub-listing of avus that includes the blessed avu) :confused-parrot:

Reply:
I double checked cyverse-de/apps#218, and it looks like it's expecting `value` to be `true` in order to set the blessed flag in the listing, so never mind that question.

Reply:
Found a bug 🐛 . When I try to set it false, I need to update metadata and not add it 🤦🏽

Reply:
so `adminSetAppAVUs` should be used instead of `adminAddAVUToApp`.

Reply:
Not a bug, it's working as intended, since multiple AVUs can be added with the same attribute. In order to update the blessed AVU, its UUID has to be included in the request: `{id, attr: BLESSED_ATTR, value, unit }`

If you want to use `adminSetAppAVUs`, then you have to include all other existing AVUs in the request (otherwise the other AVUs will be deleted).

Reply:
Now I also found a bug cos of race condition. If you try to simultaneously `blessed` and `beta` in the details dialog, `blessed` setting may or may not work 🏎️ . Gonna refactor `adminUpdateApp` to avoid this.

Reply:
Ok should be ready for review again @psarando